src/parser/Mediatype.h

```diff
@@ -0,0 +1,263 @@
+//
+//  parser/Mediatype.h
+//  librefract
+//
+//  Created by Jiri Kratochvil on 02/08/2019
+//  Copyright (c) 2019 Apiary Inc. All rights reserved.
+//
+
+#ifndef PARSER_MEDIATYPE_H
+#define PARSER_MEDIATYPE_H
+
+#include "../PEGTL/include/tao/pegtl.hpp"
+
+#include <string>
+#include <map>
+
+#include <iostream>
+
+namespace parser 
+{
+
+    namespace mediatype {
+
+        namespace pegtl = tao::pegtl;
+
+        struct restricted : pegtl::one< '!', '#', '$', '&', '^',  '_', '-', '.' >  {};
+
+        using RWS = pegtl::plus<pegtl::blank>;
+        using OWS = pegtl::star<pegtl::blank>;
+
+        // OPT: pegtl::if_must<> - if we want to throw on non-closed quotes
+        struct quoted : pegtl::seq<
+                            pegtl::one<'"'>,
+                            pegtl::plus<pegtl::not_at<pegtl::one<'"'>>, pegtl::any>,
+                            pegtl::one<'"'>
+                        > {};
+        struct ident : pegtl::seq<
```

Comment:
I don't think this matches token (from the use in key below I believe it is meant to):

>       CTL            = <any US-ASCII control character
>                        (octets 0 - 31) and DEL (127)>
>       token          = 1*<any CHAR except CTLs or separators>
>       separators     = "(" | ")" | "<" | ">" | "@"
>                      | "," | ";" | ":" | "\" | <">
>                      | "/" | "[" | "]" | "?" | "="
>                      | "{" | "}" | SP | HT

Looking at the possible octets should this allow !, +, *, ~, } and others?